syncer/syncer.go

```diff
@@ -961,6 +982,7 @@ func (s *Syncer) addJob(job *job) error {
 			// flush for every DML queue
 			metrics.AddJobDurationHistogram.WithLabelValues("flush", s.cfg.Name, s.queueBucketMapping[i], s.cfg.SourceID).Observe(time.Since(startTime).Seconds())
 		}
+		s.isTransactionEnd.Store(true)
```

Comment:
A  flush event doesn't means the transaction is ended.

Reply:
Yes. I am looking for the reason why it will fail in `handle_error_3` when removing it.